src/charm.py

```diff
@@ -622,6 +629,24 @@ def _external_url(self) -> str:
         """Return the externally-reachable (public) address of the alertmanager api server."""
         return self.model.config.get("web_external_url") or self.ingress.url or self._internal_url
 
+    @property
+    def _alertmanager_version(self) -> Optional[str]:
+        """Returns the version of Alertmanager.
+
+        Returns:
+            A string equal to the Alertmanager version.
+        """
+        container = self.unit.get_container(self._container_name)
+        if not container.can_connect():
+            return None
+        version_output, _ = container.exec([self._exe_name, "--version"]).wait_output()
+        # Output looks like this:
+        # alertmanager, version 0.23.0 (branch: HEAD, ...
+        result = re.search(r"version (\d*\.\d*\.\d*)", version_output)
+        if result is None:
+            return result
+        return result.group(1)
```

Comment:
```suggestion
        result = re.search(r"version (\d*\.\d*\.\d*)", version_output)
        return result.group(1) if result else None
```
or
```suggestion
        if (result := re.search(r"version (\d*\.\d*\.\d*)", version_output)) is not None:
            return result.group(1)
        return None
```
or
```suggestion
        if (result := re.search(r"version (\d*\.\d*\.\d*)", version_output)) is not None:
            return result.group(1)
        # None is returned implicitly
```

Reply:
The current version is :+1: from me. `.groupdict` is still more literate than the bad way Python handles result grouping from my POV, but I'm tainted from Perl dev anyway.

As a general comment (not that we're implementing any of them), can we please avoid walrus syntax wherever possible? We do it in the libraries, but as we speak of, say, a `grafana-agent` machine charm, it could probably _mostly_ share code with the existing codebase.

Following Python as they chase whatever syntax is cool these days (pattern matching, walrus, `functools....` a few years ago) saves a tiny bit of code, but also guarantees that it 💥 if we find a way to make a single `charm.py` usable in multiple cases. It's not worth the effort to strip it out yet, but we definitely don't need to be adding more outside of tests.